index.html

```diff
@@ -481,45 +485,102 @@
-      var dest  = nv[3] = (nv[3] === undefined) ? -1 : +nv[3];
-      var color = nv[4] = (nv[4] === undefined) ? "white" : nv[4];
-
-      // Update map-related variables.
-      $scope.data = maps[map]
+    $scope.players = []
+
+    /**
+     * Watch for map changes.
+     */
+    $scope.$watch("map", function(map) {
+      map = (map === undefined) ? "US" : map;
+      $scope.data = maps[map];
       $scope.cities = $scope.data.cities;
       $scope.regions = $scope.data.regions;
 
-      // Update regions to match user selection.
-      $scope.homeRegion = $scope.data.city_regions[home];
-      $scope.origRegion = $scope.data.city_regions[orig];
-      $scope.destRegion = $scope.data.city_regions[dest];
+      // Update all of the player data for the new map
+      $scope.updatePlayers();
 
-      // Update the payoff.
-      if ($scope.orig < 0 || $scope.dest < 0) {
-        $scope.payoff = "n/a";
-      } else {
-        $scope.payoff = $scope.data.payoff[orig][dest];
+      // Update the location history
+      $location.search("map", map);
+    });
+
+    /**
+     * Watch for player city changes.
+     */
+    $scope.$watch("players", function(players) {
+      $scope.updatePlayers();
+    }, true);
+
+    $scope.updatePlayers = function(players) {
+      // Update regions and payoffs.
+      for (var i = 0; i < $scope.numPlayers; i++) {
+        var player = $scope.players[i];
+        player.homeRegion = $scope.data.city_regions[player.home];
+        player.origRegion = $scope.data.city_regions[player.orig];
+        player.destRegion = $scope.data.city_regions[player.dest];
+
+        if (player.orig < 0 || player.dest < 0) {
+          player.payoff = "n/a";
+        } else {
+          player.payoff = $scope.data.payoff[player.orig][player.dest];
+        }
       }
 
-      // Update the location history.
-      $location.search({
-        map: map,
-        home: home,
-        orig: orig,
-        dest: dest,
-        color: color,
-      });
+      // Update the location history
+      $location.search("players", JSON.stringify($scope.players));
+    };
+
+    /**
+     * Watch for numPlayers changes.
+     */
+    $scope.$watch("numPlayers", function(numPlayers) {
+      $scope.updateNumPlayers();
     });
 
+    $scope.updateNumPlayers = function() {
+      var numPlayers = ($scope.numPlayers === undefined) ? 1 : +$scope.numPlayers;
+
+      // Add or remove players.
+      var playerDelta = numPlayers - $scope.players.length;
+      if (playerDelta > 0) {
+        var colorChoices = remainingColors($scope.players);
+        for (var i = 0; i < playerDelta; i++) {
+          $scope.players.push({
+            home: -1,
+            orig: -1,
+            dest: -1,
+            color: colorChoices[i],
+            payoff: "n/a",
+          });
+        }
+      } else {
+        for (var i = 0; i > playerDelta; i--) {
+          $scope.players.pop();
+        }
+      }
+    };
+
+    /**
+     * Human name --> CSS name
```

Comment:
This comment is a little strange since all of these are valid CSS color names.

http://www.w3schools.com/cssref/css_colornames.asp